src/core/utils.js

```diff
@@ -393,3 +394,42 @@ export function getClientCompatibility({
 
   return { compatible, maxVersion, minVersion, reason };
 }
+
+export function getCategoryColor(category) {
+  if (!category) {
+    throw new Error('category is required.');
+  }
+
+  if (!CATEGORY_COLORS[category.type]) {
```

Comment:
Just a nit but you could move `const maxColors = CATEGORY_COLORS[category.type];` up above this and reference `maxColors` instead.